hooks/useTransactionSender.tsx

```diff
@@ -1,113 +1,382 @@
+        }));
+      }, confirmationTimeoutMs);
+
+      subscriptionId = connection.onSignature(
+        transactionInfo.signature,
+        (result, context) => {
+          console.debug('WS result: ', transactionInfo.signature, result, context);
+          clearTimeout(timeoutId);
+          cancelSignatureSubscription(subscriptionId);
+
+          const transactionHasError = result.err !== null;
+          updateSingleTransactionInfo(transactionInfo.signature, id, () => ({
+            result: {
+              err: transactionHasError ? {} : null,
+            },
+          }));
+
+          setSuccessfulSignatureCount((state) => {
+            const current = state[id] ?? 0;
+            return {
+              ...state,
+              [id]: transactionHasError ? current : current + 1,
+            };
+          });
+        },
+        commitment,
+      );
+    });
+  };
+
+  const startProcessingTransactions = (transactionInfos: TransactionInfo<T>[]) => {
+    /**
+     * render an initial set of transactions in a single notification
+     *
+     * the `id` is a unique identifier for each notification within the
+     * mantine/core notification system. it is used to delete and update
+     * individual notifications.
+     *
+     * source: https://mantine.dev/x/notifications/#notification-props
+     */
+    const id = notifications.show({
+      title: <Text fw="bold">{generateTitle(transactionInfos)}</Text>,
+      message: generateNotficationBody(transactionInfos),
+      ...generateDefaultNotificationOptions(),
+    });
+
+    setIdToTransactionInfos((state) => ({
+      ...state,
+      [id]: transactionInfos,
+    }));
+    confirmTransactions(transactionInfos, id);
+  };
 
   const send = useCallback(
     /**
      * Sends transactions.
      * @param txs A sequence of sets of transactions. Sets are executed simultaneously.
      * @returns A sequence of set of tx signatures.
      */
-    async <T extends Transaction | VersionedTransaction>(txs: SingleOrArray<T>[]) => {
-      if (!connection || !wallet.publicKey || !wallet.signAllTransactions) {
+    async (txs: SingleOrArray<T | TransactionWithMetadata<T>>[]) => {
+      if (!connection || !wallet.publicKey || !wallet.signAllTransactions)
         throw new Error('Bad wallet connection');
-      }
-
-      if (txs.length === 0 || (txs[0] instanceof Array && txs[0].length === 0)) {
+      if (txs.length === 0 || (txs[0] instanceof Array && txs[0].length === 0))
         throw new Error('No transactions passed');
-      }
 
-      const sequence = txs[0] instanceof Array ? (txs as T[][]) : ([txs] as T[][]);
-
-      const blockhask = await connection.getLatestBlockhash();
-      const timedTxs = sequence.map((set) =>
-        set.map((e: T) => {
-          const tx = e;
-          if (!(tx instanceof VersionedTransaction)) {
-            tx.recentBlockhash = blockhask.blockhash;
-            tx.feePayer = wallet.publicKey!;
-            // Compute limit ix & priority fee ix
```

Comment:
note: these updated changes were moved to the now broken out `addComputeBudgetInstructions` function below

Reply:
Understood, is there a way to override it or rebase on top of it or just reset this PR? I added it cause I needed it out there, but otherwise we can just use this one. The only thing I added into the others was estimated costs given I tried multiple txns and fetched their compute (so just wanted them for reference in the code for others to see.